ssl/ssl_lib.c

```diff
@@ -4074,59 +4070,168 @@ int SSL_CTX_set_ct_validation_callback(SSL_CTX *ctx, ct_validation_cb callback,
             TLSEXT_TYPE_signed_certificate_timestamp)) {
         SSLerr(SSL_F_SSL_CTX_SET_CT_VALIDATION_CALLBACK,
                SSL_R_CUSTOM_EXT_HANDLER_ALREADY_INSTALLED);
-        goto err;
+        return 0;
     }
 
     ctx->ct_validation_callback = callback;
     ctx->ct_validation_callback_arg = arg;
-    ret = 1;
-err:
-    return ret;
+    return 1;
+}
+
+int SSL_ct_is_enabled(const SSL *s)
+{
+    return s->ct_validation_callback != NULL;
+}
+
+int SSL_CTX_ct_is_enabled(const SSL_CTX *ctx)
+{
+    return ctx->ct_validation_callback != NULL;
+}
+
+static void *ssl_ct_arg(int vtype)
+{
+    static int permissive = SSL_CT_VALIDATION_PERMISSIVE;
+    static int strict = SSL_CT_VALIDATION_STRICT;
+
+    switch (vtype) {
+    default:
```

Comment:
And I like to see default last because it follows natural fall-through order - which is why we should specify a rule (either yours or mine) so that the codebase looks consistent.

But since we have no rule currently (other than an example in the style guide where default happens to be last), leave as is.